tasks/request_files/request_files.py

```diff
@@ -260,23 +300,26 @@ def inner_task(
                 FILE_REQUEST_TIME_KEY: time_stamp,
                 FILE_LAST_UPDATE_KEY: time_stamp,
             }
-            if object_exists(s3, glacier_bucket, file_key):
+            file_info = get_s3_object_information(s3, glacier_bucket, file_key)
+            if file_info is not None:
+                if file_info["StorageClass"] == "DEEP_ARCHIVE" and recovery_type == "Expedited":
+                    raise RestoreRequestError(f"File '{file_key}' from bucket '{glacier_bucket}' "
```

Comment:
Instead of a raise here, wouldn't we want to populate the a_file dictionary with a failure so the updates make it to the database? This would also stop us from checking any additional files which may be ok, since an S3 bucket can utilize multiple storage types. Thoughts?

Reply:
Allowed processing of other files/granules